File: uri/estructuras/hash_tables/1256.cpp
#include <iostream>
#include <vector>
#include <sstream>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int n;
    cin >> n;
    
    bool primero = true;
    
    while (n--) {
        int m, c;
        cin >> m >> c;
        
        vector<vector<int>> tabla(m);
        
        for (int i = 0; i < c; i++) {
            int clave;
            cin >> clave;
            int indice = clave % m;
            tabla[indice].push_back(clave);
        }
        
        if (!primero) {
            cout << '\n';
        }
        primero = false;
        
        for (int i = 0; i < m; i++) {
            cout << i << " -> ";
            for (int j = 0; j < tabla[i].size(); j++) {
                cout << tabla[i][j] << " -> ";
            }
            cout << "\\\n";
        }
    }
    
    return 0;
}

File: uri/estructuras/hash_tables/1256.rs
use std::io::{self, BufRead, Write, BufWriter};

fn main() {
    let stdin = io::stdin();
    let stdout = io::stdout();
    let mut out = BufWriter::new(stdout.lock());
    let mut lines = stdin.lock().lines();
    
    let n: i32 = lines.next().unwrap().unwrap().trim().parse().unwrap();
    
    let mut primero = true;
    
    for _ in 0..n {
        let line1 = lines.next().unwrap().unwrap();
        let mut iter = line1.split_whitespace();
        let m: usize = iter.next().unwrap().parse().unwrap();
        let _c: usize = iter.next().unwrap().parse().unwrap();
        
        let line2 = lines.next().unwrap().unwrap();
        let claves: Vec<usize> = line2.split_whitespace()
            .map(|x| x.parse().unwrap())
            .collect();
        
        let mut tabla: Vec<Vec<usize>> = vec![vec![]; m];
        
        for clave in claves {
            let indice = clave % m;
            tabla[indice].push(clave);
        }
        
        if !primero {
            writeln!(out).unwrap();
        }
        primero = false;
        
        for i in 0..m {
            write!(out, "{} -> ", i).unwrap();
            for val in &tabla[i] {
                write!(out, "{} -> ", val).unwrap();
            }
            writeln!(out, "\\").unwrap();
        }
    }
}
